### ISSUE TITLE
Can't use padded_batch on dataset with distributed strategy make_dataset_iterator

### ISSUE THREAD

**Issue by lioutasb** [tensorflow/tensorflow#28941]
I can't use padded_batch when I'm trying to create a distributed iterator using the following code.

```
import tensorflow_datasets as tfds
import tensorflow as tf

tf.enable_v2_behavior()

strategy = tf.distribute.MirroredStrategy()

def encode(lang1, lang2):
    lang1 = [tokenizer_pt.vocab_size] + tokenizer_pt.encode(
        lang1.numpy()) + [tokenizer_pt.vocab_size+1]

    lang2 = [tokenizer_en.vocab_size] + tokenizer_en.encode(
        lang2.numpy()) + [tokenizer_en.vocab_size+1]

    return lang1, lang2

def tf_encode(pt, en):
    return tf.py_function(encode, [pt, en], [tf.int64, tf.int64])

with strategy.scope():
    examples, metadata = tfds.load('ted_hrlr_translate/pt_to_en', with_info=True, as_supervised=True)
    train_examples, test_examples = examples['train'], examples['test']

    tokenizer_en = tfds.features.text.SubwordTextEncoder.build_from_corpus((en.numpy() for pt, en in train_examples), target_vocab_size=10000)

    tokenizer_pt = tfds.features.text.SubwordTextEncoder.build_from_corpus((pt.numpy() for pt, en in train_examples), target_vocab_size=10000)

    train_dataset = train_examples.map(tf_encode)
    train_dataset = train_dataset.shuffle(200000)
    train_dataset = train_dataset.padded_batch(64, padded_shapes=([-1], [-1]))
    train_dataset = strategy.make_dataset_iterator(train_dataset)
```

It throws this exception:

```
ValueError: Unable to get batched dataset from the input dataset. `batch` `map_and_batch` need to be the last operations on the dataset. The batch operations can be followed by a prefetch.
```

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 1.13
- Python version: 3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 10.0/CuDNN 7.3.1
- GPU model and memory: Nvidia Titan V

**Comment by achandraa** [comment]
I was able to get the mentioned error when tried on Colab with TensorFlow GPU version 1.13.

**Comment by anj-s** [comment]
`padded_batch` should be supported. Can you try updating your tensorflow version to 1.14 and try again?

**Comment by lioutasb** [comment]
@anj-s I confirm that the issue is fixed on 1.14. Thank you.
